src/viadot/sources/eurostat.py

```diff
@@ -0,0 +1,290 @@
+                and "EUR" is the code of the specific parameter. You can add more than one parameter, but only one code per parameter!
+                So you CAN NOT provide list of codes as in example 'params = {'unit': ['EUR', 'USD', 'PLN']}'
+                These parameters are REQUIRED in most cases to pull a specific dataset from the API.
+                Both parameter and code has to be provided as a string! Defaults to None.
+            base_url (str): The base URL used to access the Eurostat API. This parameter specifies the root URL for all requests made to the API.
+                It should not be modified unless the API changes its URL scheme.
+                Defaults to "https://ec.europa.eu/eurostat/api/dissemination/statistics/1.0/data/"
+            requested_columns (List[str], optional): list of needed names of columns. Names should be given as str's into the list.
+                Defaults to None.
+            tests:
+                - `column_size`: dict{column: size}
+                - `column_unique_values`: list[columns]
+                - `column_list_to_match`: list[columns]
+                - `dataset_row_count`: dict: {'min': number, 'max', number}
+                - `column_match_regex`: dict: {column: 'regex'}
+                - `column_sum`: dict: {column: {'min': number, 'max': number}}
+        Raises:
+            APIError: If there is an error with the API request.
+            ValueError: If the resulting DataFrame is empty.
+            TypeError: If self.params is different type than a dictionary.
+            TypeError: If self.requested_columns is different type than a list containing strings.
+        Returns:
+            pd.DataFrame: Final DataFrame or raise prefect.logger.error, if issues occur.
+        """
+
+        self.dataset_code = dataset_code
+        self.params = params
+        if not isinstance(self.params, dict) and self.params is not None:
+            raise TypeError("Params should be a dictionary.")
+
+        self.base_url = f"{base_url}{self.dataset_code}?format=JSON&lang=EN"
+        self.requested_columns = requested_columns
+        if not isinstance(requested_columns, list) and requested_columns is not None:
+            raise TypeError("Requested columns should be provided as list of strings.")
+
+        # getting response from API
+        try:
+            response = handle_api_response(self.base_url, params=self.params)
+            data = response.json()
+            data_frame = self.eurostat_dictionary_to_df(["geo", "time"], data)
+        except APIError:
+            self.make_params_validation()
+
+        if data_frame.empty:
+            try:
+                self.make_params_validation()
+                self._handle_if_empty(if_empty)
+            except SKIP:
+                return pd.DataFrame()
+
+        # merging data_frame with label and last updated date
```

Comment:
This is not a strict requirement, but I would say, pick one style and stick to it throughout your code (so eg. either always uppercase or never).

IMHO it's best if the comments are written like sentences, with standard English grammar and in imperative mood (same as docstrings):

```suggestion
        # Merge data_frame with label and last updated date.
```

Reply:
Changed.